src/utils/ipv6.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <cstring>
+#include <iterator>
+#include <arpa/inet.h>
+#include <string_view>
+
+#include "utils/sugar/array_view.hpp"
+
+namespace
+{   
+    constexpr std::string_view IPV4_MAPPED_IPV6_PREFIX = "::ffff:";
+}
+
+/* Need declarations here because of -Wmissing-declarations 
```

Comment:
When Wmissing-declarations is triggered you must either put the function in a anonymous namespace / add static (.cpp) or add inline (.hpp). Pre-declaring a prototype hides future problems of multiple definitions.